docker/docker_client.go

```diff
@@ -265,7 +265,12 @@ func (c *dockerClient) setupRequestAuth(req *http.Request) error {
 		}
 		chs := parseAuthHeader(res.Header)
 		if res.StatusCode != http.StatusUnauthorized || chs == nil || len(chs) == 0 {
-			// try again one last time with Basic Auth (gcr.io for instance)
+			if res.StatusCode == http.StatusOK {
```

Comment:
I believe status != 401 would be more appropriate (meaning, I don't have a clear vision on other API endpoints)